embassy-stm32/src/wdg/mod.rs

```diff
@@ -0,0 +1,52 @@
+use core::marker::PhantomData;
+
+use embassy::util::Unborrow;
+use embassy_hal_common::unborrow;
+
+use stm32_metapac::iwdg::vals::Key;
+
+pub use stm32_metapac::iwdg::vals::Pr as Prescaler;
+
+pub struct IndependentWatchdog<'d, T: Instance> {
+    wdg: PhantomData<&'d mut T>,
+}
+
+impl<'d, T: Instance> IndependentWatchdog<'d, T> {
+    pub fn new(_instance: impl Unborrow<Target = T>, presc: Prescaler) -> Self {
```

Comment:
This should be `impl Unborrow<Target = T> + 'd`, otherwise unborrow will only live for the length of the call to `new()` and not for the entire `IndependentWatchdog` lifetime `'d`, which is unsound